wasm.go

```diff
@@ -43,65 +45,75 @@ type Runtime interface {
 	//  * Improve performance when the same module is instantiated multiple times under different names
 	//  * Reduce the amount of errors that can occur during InstantiateModule.
 	//
+	// Note: when `ctx` is nil, it defaults to context.Background.
```

Comment:
q: how do you envision ctx is used for compilation? timeout ?

Reply:
both timeout (cancel) and also tracing. For example, compilation should be in a different span. We don't have a hook yet for that, but having the context allows us to add that and have it with a coherent parent.

For example, if there's an RPC hook that compiles a module on-demand, compile can end up connected to the right trace.

Reply:
super cool